### DISCUSSION TITLE
diagnose command request using send_command inside a vdom

### DISCUSSION THREAD

**Discussion by drivero91** [vladimirs-git/fortigate-api#11] [Q&A] [answered]
Hello,

First thanks for your work on this project :) It can really help to automatize a bunch of stuffs.

My problem is that I am trying to gather information from a couple of commands inside a vdom and I have the following error:

```
from fortigate_api import FortigateAPI
fgt_api = FortigateAPI(host="10.5.54.59", username="admin", password="",vdom="XXX")
fgt_api.ssh.login()
 print(fgt_api.vdom)
fgt_api.ssh.send_command("diag sys session list | grep proto=132 -c")
# '8371: Unknown action 0\nCommand fail. Return code -1\n'
```

I think the function send_command do not know that we must apply the command inside the vdom XXX. I can apply other commands, for example _get system status_. If I try to enter the vdom via _fgt.ssh.send_command()_ I have the following error: 

```
netmiko.exceptions.ReadTimeout
Pattern not  detected 'config\\ vdom' in output.
```

I can apply the commands normally If I use the function _fgt.ssh.send_config_set()_. However, I receive all the output and not the individual stdout from each command. 

Thanks for all your help,
Daniel

**Comment by vladimirs-git** [comment, marked as answer]
In the current version (1.1.1), SSH methods do not use the 'vdom' parameter from FortigateAPI (which is only used in the REST API).
To work with a specific vdom, you need to change the vdom context by sending the commands ['config vdom', 'edit XXX'].
Please refer to the example in [ssh_vdom.py](https://github.com/vladimirs-git/fortigate-api/blob/1.1.1/examples/ssh_vdom.py).

**Reply by drivero91** [reply]
Thanks! :)
